jbmc/src/java_bytecode/java_utils.cpp

```diff
@@ -267,24 +269,32 @@ static auxiliary_symbolt declare_function(
   return func_symbol;
 }
 
-/// Create a function application expression.
+/// Create a (mathematical) function application expression.
+/// The application has the type of the codomain (range) of the function.
 /// \param function_name: the name of the function
-/// \param arguments: a list of arguments
-/// \param type: return type of the function
+/// \param arguments: a list of arguments (an element of the domain)
```

Comment:
elements of the domain

Reply:
No, the arguments as a whole are an element of the domain; each argument is an element of the projection of the domain to the member of the tuple type.